Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -1126,14 +1157,31 @@ namespace AtomSampleViewer
         }
     }
 
+    void ScriptReporter::ExportImageDiff(const char* filePath, const ScreenshotTestInfo& screenshotTestInfo)
+    {
+        using namespace AZ::Utils;
+        PngFile officialBaseline = PngFile::Load(screenshotTestInfo.m_officialBaselineScreenshotFilePath.c_str());
+        PngFile actualScreenshot = PngFile::Load(screenshotTestInfo.m_screenshotFilePath.c_str());
+
+        const size_t bufferSize = officialBaseline.GetBuffer().size();
+
+        AZStd::vector<uint8_t> diffBuffer = AZStd::vector<uint8_t>(bufferSize);
```

Comment:
Right now you are only allowing the export when the sizes match, by checking for ThresholdExceeded. This function will need some work if we ever want to allow screenshots of different size.

Reply:
Yes that was the intent. Will there ever be a case where we need to compare screenshots of different sizes?